app/controllers/stories_controller.rb

```diff
@@ -184,7 +183,8 @@ def handle_user_index
     end
     not_found if @user.username.include?("spam_") && @user.decorate.fully_banished?
     not_found unless @user.registered
-    if !user_signed_in? && (@user.spam_or_suspended? && @user.has_no_published_content?)
+    not_found if @user.spam?
+    if !user_signed_in? && (@user.suspended? && @user.has_no_published_content?)
```

Comment:
Maybe we should limit access to suspended profiles as well? Currently we show their profile in most cases (only hide for not signed in if a suspended user has content).